lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -527,7 +527,11 @@ def main():
     # otherwise the connection will fail. See https://github.com/boto/boto/issues/2836
     # for more details.
     if '.' in bucket:
-        aws_connect_kwargs['calling_format'] = OrdinaryCallingFormat()
+        # don't set OrdinaryCallingFormat for fakes3 or rgw since it is used by default
+        if s3_url and (rgw or is_fakes3(s3_url)):
+            pass
+        else:
+            aws_connect_kwargs['calling_format'] = OrdinaryCallingFormat()
```

Comment:
Thanks so much for testing this. :) That's correct, is_fakes3() looks for 'fakes3' in the url. If there is no url there is no fakes3. Just like you can't specify rgw without a url.
I'm not sure what you mean by undocumented dependency. OrdinaryCallingFormat is required for rgw and fakes3 regardless of whether or not there is a . in the bucket name which is why it's specified in get_s3_connection. It's a boto2 thing. I suppose another alternative would be to remove calling_format in the get_s3_connection function altogether and on line 531 simply have:
```
if '.' in bucket or (s3_url and (rgw or is_fakes3(s3_url))):
    aws_connect_kwargs['calling_format'] = OrdinaryCallingFormat()
```
I think I like that better, though maybe it's somewhat less readable.